src/metabase/driver/sql_jdbc.clj

```diff
@@ -122,6 +122,13 @@
         sqls       (map #(sql/format {:insert-into table-name
                                       :columns     columns
                                       :values      %})
-                        (partition-all 25 values))]
+                        (partition-all 100 values))]
+    ;; We need to partition the insert into multiple statements for both performance and correctness.
```

Comment:
curious why correctness is concerned here?

Reply:
You can only insert so many values at a time. Postgres (and I think the other DBs are the same) only allows 65,535 parameterized values.

Reply:
what does it do if we have more than that, does it throw an error or just siliently truncate it?

Reply:
By partitioning we (mostly) avoid the problem. If somebody has `> (65,534 / 100)` columns (so, 656 columns) they'll get a DB error. (Which is displayed nicely)